native-heapdump-example/README.md

```diff
@@ -0,0 +1,130 @@
+4. Create heap dumps programmatically using the [`org.graalvm.nativeimage.VMRuntime#dumpHeap`](https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/VMInspectionOptions.java) API.
+
+All approaches are described below.
+
+>Note: By default, heap dumps are created in the current working directory. The `-XX:HeapDumpPath` option can be used to specify an alternative filename or directory. For example:  
+> `./helloworld -XX:HeapDumpPath=$HOME/helloworld.hprof`
+
+>Also note: Creating heap dumps is not available on the Microsoft Windows platform.
+
+## Preparation
+
+1. Download and  install the latest GraalVM JDK with Native Image and VisualVM support using the [GraalVM JDK Downloader](https://github.com/graalvm/graalvm-jdk-downloader):
+    ```bash
+    bash <(curl -sL https://get.graalvm.org/jdk) -c 'visualvm'
+    ```
+
+2. Download the demos repository or clone it as follows:
+    ```bash
+    git clone https://github.com/graalvm/graalvm-demos
+    ```
+    ```bash
+    cd graalvm-demos/native-heapdump-example
+    ```
+
+## Dump the Initial Heap of a Native Executable
+
+Use the `-XX:+DumpHeapAndExit` command-line option to dump the initial heap of a native executable.
+This can be useful to identify which objects the Native Image build process allocated to the executable's heap. 
+For a HelloWorld example, use the option as follows:
+
+```shell
+$GRAALVM_HOME/bin/native-image HelloWorld --enable-monitoring=heapdump
+./helloworld -XX:+DumpHeapAndExit
+Heap dump created at '/path/to/helloworld.hprof'.
+```
+
+## Create Heap Dumps with SIGUSR1 (Linux/macOS only)
+
+>Note: This requires the `Signal` API, which is enabled by default except when building shared libraries.
+
+The following example is a simple multi-threaded Java application that runs for 60 seconds. 
+This provides you with enough time to send it a `SIGUSR1` signal. The application will handle the signal and create a heap dump in the application's working directory. The heap dump will contain the `Collection` of `Person`s referenced by the static variable `CROWD`.
+
+Follow these steps to build a native executable that will produce a heap dump when it receives a `SIGUSR1` signal.
+
+### Build the Application
+
+1. Compile _SVMHeapDump.java_ using `javac`:
+    ```bash
+    $JAVA_HOME/bin/javac SVMHeapDump.java
+    ```
+
+2. Build a native executable using the `--enable-monitoring=heapdump` command-line option:
+    ```shell
+    $JAVA_HOME/bin/native-image SVMHeapDump --enable-monitoring=heapdump
+    ```
+    This causes the resulting native executable to produce a heap dump when it receives a `SIGUSR1` signal.
+
+    The `native-image` builder creates a native executable from the `SVMHeapDump.class`.
+    When the command completes, the native executable `svmheapdump` is created in the current directory.)
+
+### Run the Applicatoin
+
+1. Run the application send it a signal, and check the heap dump.:
```

Comment:
Run the application send it a signal, and check the heap dump.: >  Run the application, send it a signal, and check the heap dump: